bluemira/equilibria/optimisation/harmonics/harmonics_approx_functions.py

```diff
@@ -110,7 +110,7 @@ def coil_harmonic_amplitude_matrix(
         / np.sqrt(degrees * (degrees + 1))
     )
 
-    return currents2harmonics
+    return np.array(currents2harmonics, dtype=np.float32)
```

Comment:
I think downcasting is wrong here. It will lead to hard to track down problems elsewhere because the arrays will then be upcast to float64 as some point. 

![image](https://github.com/user-attachments/assets/debcdf59-2127-47c3-a2b9-15ead3047291)

Reply:
Thanks @je-cook - I have removed this and round instead.